menoh/mkldnn_with_generic_fallback/backend/mkldnn/mkldnn_context.cpp

```diff
@@ -0,0 +1,163 @@
+                                  if(found != table.end()) {
+                                      assert(found->second.dims().size() == 2 ||
+                                             found->second.dims().size() == 4);
+                                      input_list.push_back(found->second);
+                                      return true;
+                                  }
+                                  return false;
+                              };
+                            if(found_from_common_table(
+                                 common_parameter_table)) {
+                                *logger << input_name
+                                        << " is found in common parameter table"
+                                        << std::endl;
+                                break;
+                            }
+                            if(found_from_common_table(common_input_table)) {
+                                *logger << input_name
+                                        << " is found in common input table"
+                                        << std::endl;
+                                break;
+                            }
+
+                            // search in other contexts' variable table
+                            bool is_found_from_other_context = false;
+                            for(auto const& context_pair : context_list) {
+                                if(context_pair.first == context_name) {
+                                    continue; // skip self
+                                }
+                                auto found =
+                                  context_pair.second->try_to_get_variable(
+                                    input_name);
+                                if(found) {
+                                    *logger << input_name
+                                            << " is found in other context's "
+                                               "varibale table: "
+                                            << context_pair.first << std::endl;
+                                    procedure copy_proc; // copy from other
+                                                         // context to array
+                                    array arr;
+                                    std::tie(copy_proc, arr) = *found;
+                                    assert(arr.dims().size() == 2 ||
+                                           arr.dims().size() == 4);
+                                    new_copy_procedure_list.push_back(
+                                      copy_proc);
+                                    input_list.push_back(arr);
+                                    auto format =
+                                      arr.dims().size() == 2
+                                        ? mkldnn::memory::format::nc
+                                        : mkldnn::memory::format::nchw;
```

Comment:
It seems that the `format` variable is unused.

Reply:
I see. I'll fix.